spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindNullDeref.java

```diff
@@ -1919,6 +1920,21 @@ private boolean isGeneratedCodeInCatchBlockViaLineNumber(@NonNull ConstantPool c
                     }
                 }
                 break;
+            case Const.INVOKEINTERFACE:
+                if (handle.getInstruction() instanceof INVOKEINTERFACE) {
+                    String methodName = ((INVOKEINTERFACE) handle.getInstruction()).getMethodName(cpg);
+                    if ("close".equals(methodName)) {
+                        // the close methods get the line number of the end of the try block assigned
+                        if (throwables.stream().anyMatch(x -> lineNumberTable.getSourceLine(x.getEndPC()) == currentLine)) {
+                            closeCounter++;
+                        }
+                    } else if ("addSuppressed".equals(methodName)) {
+                        if (relevantLineNumbers.contains(currentLine)) {
+                            addSuppressedPresent = true;
+                        }
+                    }
+                }
+                break;
```

Comment:
Even though I remove these lines, the `./gradlew build` was completed without test failure. Does it mean that the test case you added is not enough to reproduce the issue?

Reply:
Let me check ,it was important whether test is compiled with debug symbols or without them

Reply:
if possible, configure Gradle's JavaCompile tasks to make the test reproducible.

Reply:
I think I was able to reproduce it before via spotbugs tests, but maybe something changed there.
For now can reproduce in following environment:
```console
$ gradle --version

------------------------------------------------------------
Gradle 7.3
------------------------------------------------------------

Build time:   2021-11-09 20:40:36 UTC
Revision:     96754b8c44399658178a768ac764d727c2addb37

Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.1 (Oracle Corporation 17.0.1+12-nixos)
OS:           Linux 5.10.81 amd64
```
```console
$ java --version
openjdk 11.0.9 2020-10-20
OpenJDK Runtime Environment (build 11.0.9+0-adhoc..source)
OpenJDK 64-Bit Server VM (build 11.0.9+0-adhoc..source, mixed mode)
```
by running
```console
$ git checkout origin/master spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindNullDeref.java
$ gradle assemble
$ java -cp "./spotbugs/build/libs/spotbugs.jar:./spotbugs-annotations/build/libs/*:./spotbugs/.libs/*"  edu.umd.cs.findbugs.LaunchAppropriateUI -textui -longBugCodes -onlyAnalyze "ghIssues.issue600.*" spotbugsTestCases/build/libs/spotbugsTestCases-*.jar
M C RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE RCN: Nullcheck of x at line 7 of value previously dereferenced in ghIssues.issue600.TryWithResourcesSimple.nocatchInterface(IMyAutocloseable)  At TryWithResourcesSimple.java:[line 6]
...

can try to debug why `gradle test` doesn't fail on new test without the fix, but going to take a bit of time

Reply:
invocation for just issue600 classes
> java -cp "./spotbugs/build/libs/spotbugs.jar:./spotbugs-annotations/build/libs/*:./spotbugs/.libs/*"  edu.umd.cs.findbugs.LaunchAppropriateUI -textui -longBugCodes -onlyAnalyze "ghIssues.issue600.*" -auxclasspath spotbugsTestCases/build/classes/java/main/ghIssues/issue600/MyAutocloseable.class:spotbugsTestCases/build/classes/java/main/ghIssues/issue600/IMyAutocloseable.class  spotbugsTestCases/build/classes/java/main/ghIssues/issue600/TryWithResourcesSimple.class 
>
> M C RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE RCN: Nullcheck of x at line 7 of value previously dereferenced in ghIssues.issue600.TryWithResourcesSimple.nocatchInterface(IMyAutocloseable)  At TryWithResourcesSimple.java:[line 6]

Reply:
`RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` before vs `RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE` now, will update test to account for that

Reply:
Updated the test, now it fails for me without the fix

Reply:
I also confirmed that `./gradlew build` fails if I remove these lines.
Thanks!